pkg/virtctl/expand/expand.go

```diff
@@ -0,0 +1,152 @@
+)
+
+func ExpandCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd = &cobra.Command{
+		Use:     "expand (VM)",
+		Short:   "Expands a virtual machine.",
+		Example: usage(COMMAND_EXPAND),
+		Args:    templates.ExactArgs("vm", 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			expand := Command{clientConfig: clientConfig}
+			return expand.Run(args)
+		},
+	}
+	cmd.Flags().BoolVarP(&filePathProvided, fileFlag, fileFlagShort, false, "If present, the Virtual Machine spec in provided file will be expanded")
+	cmd.Flags().StringVarP(&namespaceProvided, namespaceFlag, namespaceFlagShort, "", "If present, the namespace scope for this CLI request")
+	cmd.Flags().StringVarP(&outputProvided, outputFlag, outputFlagShort, YAML, "Specify a format that will be used to display output.")
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	return cmd
+}
+
+func usage(cmd string) string {
+	usage := fmt.Sprintf("  # %s a virtual machine called 'myvm':\n", cmd)
+	usage += fmt.Sprintf("  {{ProgramName}} %s kubevirt myvm\n", cmd)
+	return usage
+}
+
+func createVMObject(filePath string) (*v1.VirtualMachine, error) {
+	readFile, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(readFile), 1024)
+	vm := &v1.VirtualMachine{}
+	err = decoder.Decode(vm)
+	if err != nil {
+		return nil, err
+	}
+
+	return vm, nil
+}
+
+func printExpandedVM(outputFormat string, expandedVm *v1.VirtualMachine) error {
+	var formatedOutput []byte
+	var err error
+
+	switch outputFormat {
+	case JSON:
+		formatedOutput, err = json.MarshalIndent(expandedVm, "", " ")
+	case YAML:
+		formatedOutput, err = yamlv3.Marshal(expandedVm)
+	default:
+		return fmt.Errorf("not supported output format defined: %v", outputFormat)
+	}
+
+	if err != nil {
+		return err
+	}
+
+	cmd.Print(string(formatedOutput))
+	return nil
+}
+
+type Command struct {
+	clientConfig clientcmd.ClientConfig
+}
+
+func expandVirtualMachine(virtClient kubecli.KubevirtClient, args []string) error {
+	var expandedVm *v1.VirtualMachine
+	var err error
+
+	if filePathProvided {
+		vm, err := createVMObject(args[0])
+		if err != nil {
+			return err
+		}
+
+		expandedVm, err = virtClient.ExpandSpec(vm.Namespace).ForVirtualMachine(vm)
+		if err != nil {
+			return fmt.Errorf("error expanding VirtualMachine %s: %v", vm.Name, err)
+		}
+	} else {
+		vmName := args[0]
+		expandedVm, err = virtClient.VirtualMachine(namespaceProvided).GetWithExpandedSpec(vmName)
+		if err != nil {
+			return fmt.Errorf("error expanding VirtualMachine %s: %v", vmName, err)
+		}
+	}
+
+	err = printExpandedVM(outputProvided, expandedVm)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (cmdExpand *Command) Run(args []string) error {
+	virtClient, err := kubecli.GetKubevirtClientFromClientConfig(cmdExpand.clientConfig)
+	if err != nil {
+		return fmt.Errorf("cannot obtain KubeVirt client: %v", err)
+	}
+
+	if namespaceProvided == "" {
```

Comment:
As mentioned above you should use the namespace flag from `virtctl` and the namespace should already been set and this isn't needed